pkg/operator/controller/gateway-service-dns/controller_test.go

```diff
@@ -246,12 +266,54 @@ func Test_Reconcile(t *testing.T) {
 			expectUpdate: []client.Object{},
 			expectDelete: []client.Object{},
 		},
+		{
+			name: "gateway with two unique host names, one of which clashes with an existing ingress controller",
+			existingObjects: []runtime.Object{
+				dnsConfig, infraConfig,
+				gw(
+					"example-gateway",
+					l("stage-https", "*.stage.apps.example.com", 443),
+					l("apps-https", "*.apps.example.com", 443),
+				),
+				svc("example-gateway", gatewayManagedLabel, exampleGatewayLabel, ingHost("lb.example.com")),
+				ic("default", "apps.example.com"),
+			},
+			reconcileRequest: req("openshift-ingress", "example-gateway"),
+			expectCreate: []client.Object{
+				dnsrecord("example-gateway-644bf77744-wildcard", "*.stage.apps.example.com.", iov1.ManagedDNS, exampleGatewayLabel, "lb.example.com"),
+			},
+			expectUpdate: []client.Object{},
+			expectDelete: []client.Object{},
+		},
+		{
+			name: "gateway with two unique host names, neither of which clashes with an existing ingress controller",
+			existingObjects: []runtime.Object{
+				dnsConfig, infraConfig,
+				gw(
+					"example-gateway",
+					l("stage-https", "*.stage.apps.example.com", 443),
+					// apps.example.com looks like it'll clash with the default ic below, but the ic creates a record
+					// for *.apps.example.com, which doesn't actually clash with apps.example.com
+					l("apps-https", "apps.example.com", 443),
```

Comment:
This should clash though, because the DNSRecord for the listener will prefix it with `*.` too.